common/shell/shell_platform.c

```diff
@@ -0,0 +1,607 @@
+        gpio_idx_in_group = gpio_idx;
+    } else if (mode == CHECK_BY_GLOBAL_IDX) {
+        gpio_idx_in_group = gpio_idx%GPIO_GROUP_SIZE;
+    } else
+        return 1;
+
+    if ( (cfg->port_pin_mask & (gpio_port_pins_t)BIT(gpio_idx_in_group) ) == 0U)
+        return 1;
+
+    return 0;
+}
+#endif
+
+static int gpio_access_cfg(const struct shell *shell, int gpio_idx, enum GPIO_ACCESS mode, int *data)
+{
+    if (!shell)
+        return 1;
+
+    if (gpio_idx >= GPIO_CFG_SIZE || gpio_idx < 0)
+        return 1;
+
+    /* check whether reserve pin defined in pin NAME */
+    /*
+    if ( !strncmp(GPIO_RESERVE_PREFIX, gpio_name[gpio_idx], 7) )
+        return 1;
+    */
+    switch (mode)
+    {
+    case GPIO_READ:
+        if ( gpio_cfg[gpio_idx].is_init == DISABLE )
+            return 1;
+
+        uint32_t g_val = sys_read32(GPIO_GROUP_REG_ACCESS[gpio_idx/32]);
+        uint32_t g_dir = sys_read32(GPIO_GROUP_REG_ACCESS[gpio_idx/32]+0x4);
+
+        char *pin_prop = (gpio_cfg[gpio_idx].property == OPEN_DRAIN) ? "OD":"PP";
+        char *pin_dir = (gpio_cfg[gpio_idx].direction == GPIO_INPUT) ? "input":"output";
+
+        char *pin_dir_reg = "I";
+        if ( g_dir & BIT(gpio_idx%32) )
+            pin_dir_reg = "O";
+
+        int val = gpio_get(gpio_idx);
+        if (val == 0 || val == 1)
+            shell_print(shell, "[%-3d] %-35s: %-3s | %-6s(%s) | %d(%d)",
+                gpio_idx, gpio_name[gpio_idx], pin_prop, pin_dir, pin_dir_reg, val, GET_BIT_VAL(g_val, gpio_idx%32));
+        else
+            shell_print(shell, "[%-3d] %-35s: %-3s | %-6s(%s) | %s",
+                gpio_idx, gpio_name[gpio_idx], pin_prop, pin_dir, pin_dir_reg, "resv");
+        
+        break;
+
+    case GPIO_WRITE:
+        if (!data)
+            return 1;
+
+        if ( *data!=0 && *data!=1 )
+            return 1;
+
+        if ( gpio_set(gpio_idx, *data) )
+            return 1;
+            
+        break;
+    
+    default:
+        LOG_ERR("No such mode %d!", mode);
+        break;
+    }
+    
+    return 0;
+}
+
+static int gpio_get_group_idx_by_dev_name(const char *dev_name)
+{
+    if (!dev_name)
+        return -1;
+
+    int group_idx = -1;
+    for (int i=0; i<ARRAY_SIZE(GPIO_GROUP_NAME_LST); i++) {
+        if (!strcmp(dev_name, GPIO_GROUP_NAME_LST[i]))
+            group_idx = i;
+    }
+
+    return group_idx;
+}
+
+static const char* gpio_get_name(const char *dev_name, int pin_num)
+{
+    if (!dev_name)
+        return NULL;
+
+    int name_idx = -1;
+    name_idx = pin_num + 32 * gpio_get_group_idx_by_dev_name(dev_name);
+
+    if (name_idx == -1)
+        return NULL;
+
+    if (name_idx >= GPIO_CFG_SIZE)
+        return "Undefined";
+
+    return gpio_name[name_idx];
+}
+#endif
+
+/* 
+    Command SENSOR 
+*/
+#if COMMAND_SENSOR == 1
+static bool access_check(uint8_t sensor_num) {
+    bool (*access_checker)(uint8_t);
+    access_checker = sensor_config[SnrNum_SnrCfg_map[sensor_num]].access_checker;
```

Comment:
This function and many that follow need to use a standardized naming for sensor as it uses "sensor", Snr", "sen" and "snr"

Reply:
Got it, this suggestion would add in next pr, thanks!